generated/nidcpower/nidcpower/session.py

```diff
@@ -6911,11 +6911,11 @@ def __init__(self, resource_name, channels=None, reset=False, options={}, indepe
         options = _converters.convert_init_with_options_dictionary(options)
 
         # Call specified init function
-        # Note that _library_interpreter sets _vi to 0 in its constructor, so that if
+        # Note that _interpreter clears the session handle in its constructor, so that if
         # _fancy_initialize fails, the error handler can reference it.
-        # And then once _fancy_initialize succeeds, we can update _library_interpreter._vi
-        # with the actual session handle.
-        self._interpreter._vi = self._fancy_initialize(resource_name, channels, reset, options, independent_channels)
+        # And then once _fancy_initialize succeeds, we can call this again with the
+        # actual session handle.
+        self._interpreter.set_session_handle(self._fancy_initialize(resource_name, channels, reset, options, independent_channels))
```

Comment:
This change was to stop breaking encapsulation, right?
I like it overall. It may even be worth calling out in the PR description.

The comment now feels a little murky, though. `we can call this again`. Is it referring to the specified init function that the first line of the comment mentions calling or to set_session_handle? From context clues, I would say the latter, but the comment could be clearer by saying `we can call set_session_handle again`.

Reply:
Not entirely, no. It was primarily made so that Session.close() can reset the session handle without having to know that it's an int for LibraryInterpreter but a session_pb2.Session for GrpcStubInterpreter. With the udpated code, now Session.close just calls set_session_handle() without specifying any handle, so that the default value gets used to reset it.

I'll clean up the comment.

Reply:
Done